lib/rubocop/cop/lint/hash_missing_class.rb

```diff
@@ -0,0 +1,80 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Lint
+      # Checks for `hash` methods that delegate to `Array#hash` without including
+      # `self.class`. Including `self.class` ensures instances of different classes
+      # have distinct hash values, which is generally expected.
+      #
+      # @safety
+      #   This cop is unsafe because it can produce false positives if the class
+      #   intentionally wants instances with the same state to hash equally
+      #   regardless of their class.
```

Comment:
If this kind of case is considered false positives, I think it should be categorized as a `Style` cop rather than a `Lint` cop.
If the concern is not about incorrect detection but only about changes in behavior, then it should be marked as `SafeAutoCorrect: false`, not `Safe: false`.

Reply:
Yeah, looking the the Ruby issue that prompted the creation of the cop this also sounds like a Style, not a Lint to me. 

> When implementing your own #hash based on multiple values, the best practice is to combine the class and any values using the hash code of an array: